ocs_ci/utility/powernodes.py

```diff
@@ -56,6 +57,37 @@ def verify_machine_is_down(self, node):
         else:
             return True
 
+    def stop_powernodes_machines_powervs(
```

Comment:
Instead of power class, it can be its own class `kube_service.py` , I think in general its applicable for all platforms and should be different category than node reboots.

Reply:
Vasu, thanks for your comments. But I am aware of only code plugged into platform_nodes. If I implement this into a separate class, how do I link it to platform_nodes? Test cases get an instance of platform node and invoke node restarts from platform_nodes interface.

Reply:
I will check the test case  to see how we can keep the distinction clear.

Reply:
@gitsridhar ,
If I understand your requirements properly, you need control over node and service. Why can't we have a composition here. 
```python

class Service(object):
   self.service_type=None

class Kubelet(Service):
  def __init__(self):
    super().__init__("kubelet")
 
  def start():
  def stop():
  def kill(graceful=True):     

class IBMPowerNodes(NodeBase):
   
  def __init_(self): 
    service = None

  def init_service_obj(self, service_name):
     service = service_class()
```

From the test we can do:
```python
node = IBMPowerNodes()
node.init_service_obj('kubelet')
node.service.start()
node.service.stop()
node.service.kill()
```

We can rethink the class in case if you need multiple service types to be controlled.

Not sure if this helps in your case .
Thoughts ?

Reply:
@gitsridhar , sorry I didn't realize that you are already doing something similar.

Reply:
@gitsridhar I agree with Vasu to move this to some other file as stop/restart/start of service can be used not just on power nodes.

So you can move just parts from here about start/stop/restart to some service.py file and describe that this action happens via ssh.  The rest of the logic can stay in this file as you have.

Reply:
Petr and Shylesh, thanks for your quick comments. I will amend this PR shortly based on your comments.

I am doing this based on one observation: when I manually ssh into a storage node and stop kubelet service, the node moves to the NotReady state and in about 5 minutes, all pods running on that node get terminated and either relocated to another node or moves to the pending state. When kubelet service is started, pods get back to the original Ready state. I understand this is equivalent to shutting down the entire storage node and starting it back after few minutes by ocs-ci disruptive test cases. Please comment if this understanding is not correct.

Reply:
@shylesh I am doing only part of what you said (tie into IBMPowerNodes..) but I agree with you on class separation, so it can be used by anyone for any service.

Reply:
@shylesh @petr-balogh @vasukulkarni I just pushed an amend to PR, I have addressed all of your comments.